src/main/java/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -210,4 +212,32 @@ fun storeCancellableContinuation(cont: CancellableContinuation<*>) {
     } else {
         storedLastCancellableCont = cont
     }
-}
\ No newline at end of file
+}
+
+object UnsafeHolder {
+    @Volatile
+    private var theUnsafe: Unsafe? = null
+
+    @JvmStatic
+    fun getUnsafe(): Unsafe {
+        if (theUnsafe == null) {
+            try {
+                val f = Unsafe::class.java.getDeclaredField("theUnsafe")
+                f.isAccessible = true
+                theUnsafe = f.get(null) as Unsafe
+            } catch (e: Exception) {
+                throw RuntimeException(e)
+            }
+        }
+
+        return theUnsafe!!
+    }
+}
+
+fun collectThreadDump() = Thread.getAllStackTraces()
```

Comment:
Let's move the filtering logic in `Reporter` into this method (it is better when all the thread dump transformations are in a single place)